### DISCUSSION TITLE
LazyView

### DISCUSSION THREAD

**Discussion by Senso4sLab** [xamarin/XamarinCommunityToolkit#1030] [Q&A] [answered]
I tried to use ControlTemplate inside LazyView control. Unfortunately, it does not work.  Can someone explain what I have to change in class BaseLazyView. 

Thanks in advance

**Comment by pictos** [comment, marked as answer]
@Senso4sLab you can't use ControlTemplate with LazyView, you need to pass the view's type. You can take a look at our sample to have a better idea of how to use it.

https://github.com/xamarin/XamarinCommunityToolkit/blob/develop/samples/XCT.Sample/Pages/Views/TabView/LazyTabPage.xaml

**Comment by pictos** [comment]
I received those questions in my email, so I'll reply to those here to make it more useful for all devs.

> In general, it works fine but I would like to extend this functionality to be able to use TemplateControl. According to your reply, it seems that this is not possible to do? 

Yes, LazyView doesn't support TemplateControl(TC) right now, if we can make it work with TC we can implement this feature.

> Can be as kind as possible and explain in more detail what is the problem?

The LazyView concept wasn't my idea, I took it from [this article](https://www.sharpnado.com/xamarin-forms-lazyview-boost-your-app-reactivity-and-startup-time/). I believe that it doesn't implement TC because it can't easily be lazy load. I know that on Shell we can pass a page as ControlTemplate and that will make the page be lazy load, which means, the page will just be allocated when the user navigates to it. But for controls that aren't true (AFAIK), so if you have something like this

```xml
<ContentPage .../>
<StackLayout>
    <Button Text="Load something ruge" Command="{Binding ClickCommand}" />
     <LazyView ControlTemplate={local:MyView} />
</StackLayout>
</ContentPage>
```
In that case, the LazyView will be allocated right after the page is allocated. Using the `TypeArguments` we're providing the type of the `View` that we want to renderer and we can allocate that `View` when we want to.

And if you or anyone finds a way to implement a LazyView with TC just open an issue with a spec and send a PR❣

> I know that input to LazyView is Views class and its type is defined with TypeArguments. Can you give me some hints in which direction I have to go to extend the current functionality?

In the article that I provide, you can find another implementation, that you can use as a sample. The short answer will be, just create a class that extends from `BaseLazyView` and you will be able to create your implementation of LazyView. There's an XCT show to be out soon about LazyViews, I'll update this answer with the link ASAP, there I showed a little bit about how you can extend the LazyView.

**Reply by Senso4sLab** [reply]
Thank for the explanation.
